packages/lodestar/src/chain/chain.ts

```diff
@@ -257,38 +258,83 @@ export class BeaconChain implements IBeaconChain {
     });
   }
 
-  getForkDigest(): ForkDigest {
-    const state = this.getHeadState();
-    return computeForkDigest(this.config, state.fork.currentVersion, this.genesisValidatorsRoot);
+  getHeadForkDigest(): ForkDigest {
+    return this.getForkDigest(this.getHeadForkName());
+  }
+  getClockForkDigest(): ForkDigest {
+    return this.getForkDigest(this.getClockForkName());
   }
 
-  getForkName(): ForkName {
-    return computeForkNameFromForkDigest(this.config, this.genesisValidatorsRoot, this.getForkDigest());
+  getForkDigest(forkName: ForkName): phase0.ForkDigest {
+    const forkDigest = this.forkDigestCache.get(forkName);
+    if (forkDigest) return forkDigest;
+    throw new Error("No forkDigest for " + forkName);
   }
 
-  getENRForkID(): phase0.ENRForkID {
-    const state = this.getHeadState();
-    const currentVersion = state.fork.currentVersion;
+  getHeadForkName(): ForkName {
+    return this.config.getForkName(this.getHeadState().slot);
+  }
+
+  getClockForkName(): ForkName {
+    return this.config.getForkName(this.clock.currentSlot);
+  }
 
-    const forkDigest = this.getForkDigest();
+  getForkName(forkDigest: ForkDigest): ForkName {
+    for (const [name, value] of this.forkDigestCache.entries()) {
+      if (byteArrayEquals(forkDigest as Uint8Array, value as Uint8Array)) {
+        return name;
+      }
+    }
+    throw new Error("No fork name for fork digest" + toHexString(forkDigest));
+  }
+
+  getENRForkID(): phase0.ENRForkID {
+    const {currentFork, nextFork} = this.getForks();
 
     return {
-      forkDigest,
-      // TODO figure out forking
-      nextForkVersion: currentVersion.valueOf() as Uint8Array,
-      nextForkEpoch: FAR_FUTURE_EPOCH,
+      forkDigest: this.getHeadForkDigest(),
+      nextForkVersion: nextFork ? nextFork.version : currentFork.version,
+      nextForkEpoch: nextFork ? nextFork.epoch : FAR_FUTURE_EPOCH,
     };
   }
 
   getStatus(): phase0.Status {
     const head = this.forkChoice.getHead();
     const finalizedCheckpoint = this.forkChoice.getFinalizedCheckpoint();
     return {
-      forkDigest: this.getForkDigest(),
+      forkDigest: this.getHeadForkDigest(),
       finalizedRoot: finalizedCheckpoint.epoch === GENESIS_EPOCH ? ZERO_HASH : finalizedCheckpoint.root,
       finalizedEpoch: finalizedCheckpoint.epoch,
       headRoot: head.blockRoot,
       headSlot: head.slot,
     };
   }
+
+  /**
+   * Precompute current fork and next fork if it's available
+   */
+  private initForkDigestCache(): void {
```

Comment:
I think we should prevent adding methods to a class when they are not strictly necessary. You can have a standalone pure function like:
```ts
constructor() {
  this.forkDigestByForkName = computeForkDigestByForkName(config, genesisValidatorsRoot, forks)
}

function computeForkDigestByForkName() {}
```
This way it's very obvious that we will never call the equivalent of `initForkDigestCache` again. Also super easy to unit test